Sources/TuistKit/Services/CacheService.swift

```diff
@@ -1,27 +1,45 @@
 import Foundation
 import TSCBasic
+import TuistCore
+import TuistLoader
 import TuistSupport
 
 final class CacheService {
     /// Cache controller.
     private let cacheController: CacheControlling
 
-    init(cacheController: CacheControlling = CacheController()) {
+    /// Generator Model Loader, used for getting the user config
+    private let generatorModelLoader: GeneratorModelLoader
+
+    init(cacheController: CacheControlling = CacheController(),
+         manifestLoader: ManifestLoader = ManifestLoader(),
+         manifestLinter: ManifestLinter = ManifestLinter()) {
         self.cacheController = cacheController
+        generatorModelLoader = GeneratorModelLoader(manifestLoader: manifestLoader,
+                                                    manifestLinter: manifestLinter)
     }
 
     func run(path: String?) throws {
         let path = self.path(path)
-        try cacheController.cache(path: path)
+        let userConfig = try loadConfig()
+        try cacheController.cache(path: path, userConfig: userConfig)
     }
 
     // MARK: - Helpers
 
     private func path(_ path: String?) -> AbsolutePath {
         if let path = path {
-            return AbsolutePath(path, relativeTo: FileHandler.shared.currentPath)
+            return AbsolutePath(path, relativeTo: currentPath)
         } else {
-            return FileHandler.shared.currentPath
+            return currentPath
         }
     }
+
+    private func loadConfig() throws -> Config {
```

Comment:
I think having a method that calls another method is a bit too much. They don't make the code more readable either so I'd just remove them.

Reply:
updated.